Src/Web/Web.Shared.Net/OperationCorrelationTelemetryInitializer.cs

```diff
@@ -90,16 +90,23 @@ public OperationCorrelationTelemetryInitializer()
             
             if (string.IsNullOrEmpty(requestTelemetry.Source) && currentRequest.Headers != null)
             {
-                string sourceIkey = currentRequest.Headers[RequestResponseHeaders.SourceInstrumentationKeyHeader];
+                string sourceAppId = currentRequest.Headers[RequestResponseHeaders.SourceAppIdHeader];
+
+                string myAppId = string.Empty;
+                bool foundMyAppId = false;
+                if (!string.IsNullOrEmpty(requestTelemetry.Context.InstrumentationKey))
+                {
+                    foundMyAppId = CorelationIdLookupHelper.TryGetAppId(requestTelemetry.Context.InstrumentationKey, out myAppId);
```

Comment:
Out of curiosity. Why this syntax with an output parameter vs returning the id or null if not found?